stl/inc/ranges

```diff
@@ -3281,12 +3281,21 @@ namespace ranges {
             friend _Outer_iter<true>;
 
             using _Mybase   = _Outer_iter_base<iterator_t<_Maybe_const<_Const, _Vw>>>;
-            using _ParentTy = _Maybe_const<_Const, split_view>;
+            using _ParentTy = _Maybe_const<_Const, lazy_split_view>;
             using _BaseTy   = _Maybe_const<_Const, _Vw>;
 
-            _ParentTy* _Parent = nullptr;
+            _ParentTy* _Parent   = nullptr;
+            bool _Trailing_empty = false;
 
-            _NODISCARD constexpr iterator_t<_BaseTy>& _Get_current() noexcept {
+            _NODISCARD constexpr iterator_t<_BaseTy> _Get_current() && noexcept {
+                if constexpr (forward_range<_BaseTy>) {
+                    return _STD move(this->_Current);
+                } else {
+                    return _STD move(*_Parent->_Current);
+                }
+            }
+
+            _NODISCARD constexpr iterator_t<_BaseTy>& _Get_current() & noexcept {

```

Comment:
The tests are failing with:
```
In file included from C:\a\1\s\tests\std\tests\P0896R4_ranges_alg_mismatch\test.cpp:8:
D:\build\out\inc\ranges(3306,61): error: cannot overload a member function without a ref-qualifier with a member function with ref-qualifier '&'
            _NODISCARD constexpr const iterator_t<_BaseTy>& _Get_current() const noexcept {
                                                            ^
D:\build\out\inc\ranges(3298,55): note: previous declaration is here
            _NODISCARD constexpr iterator_t<_BaseTy>& _Get_current() & noexcept {
                                                      ^
D:\build\out\inc\ranges(3306,61): error: cannot overload a member function without a ref-qualifier with a member function with ref-qualifier '&&'
            _NODISCARD constexpr const iterator_t<_BaseTy>& _Get_current() const noexcept {
                                                            ^
D:\build\out\inc\ranges(3290,54): note: previous declaration is here
            _NODISCARD constexpr iterator_t<_BaseTy> _Get_current() && noexcept {
                                                     ^
2 errors generated.
```

Reply:
Oh I will just add this tiny little overload, what could possibly go wrong?

Reply:
Fixed and simplified